### PR TITLE
feat(benchmarks): add largetable benchmark and refactor tree benchmarks

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/benchmarks/e2e_test/tree_perf.ts [hunk 1]
```diff
@@ -26,6 +26,13 @@ describe('tree benchmark perf', () => {
     }).then(done, done.fail);
   });
 
+  fit('should run for ng2 switch', (done) => {
```
Comment (vicb) [inline modules/benchmarks/e2e_test/tree_perf.ts]: debug

**Comment by vicb** [comment]
CI fails

**Comment by tbosch** [comment]
@vicb I added more benchmarks, could you take another look?

**Comment by vicb** [comment]
👍

**Comment by mprobst** [comment]
Merged as df4c0a3d1fdb5982e6472c5ee1db2639d28c08b0.

**Pull request closed without merging**